### DISCUSSION TITLE
Nested list margin-bottom seems not honored

### DISCUSSION THREAD

**Discussion by aschrijver** [tailwindlabs/tailwindcss#5120] [Help] [answered]
I am porting a Jekyll site from Boostrap to Tailwind and in the markdown rendering I notice that in a multi-level Table of Contents the nested list has a larger margin. According to the DEFAULT typography settings I would expect this not to be the case.

So the markdown looks like this:

```markdown
- [Introduction](#introduction)
- [Problem](#problem)
- [Context](#context)
- [Discussion](#discussion)
  - [Community defined](#community-defined)
  - [What ActivityPub offers](#what-activitypub-offers)
- [Solution](#solution)
  - [Domain model](#domain-model)
  - [Vocabulary extension](#vocabulary-extension)
```

The jekyll `post.html` layout has this html:

```html
    <section class="md:w-3/4 prose md:prose-lg lg:prose-xl">
      {{ content }}
    </section>
```

And here you see what's rendered, and the `0.75rem` margin-bottom that is overruled by the last `<li>` of the nested list:

![tailwind-nested-list-margin](https://user-images.githubusercontent.com/5111931/127563148-76072c65-33a8-45d5-ba51-0bad16c897ee.png)

Note that the total space above "Solution" is too large, as the `1.25rem` is used instead.

Is this how it should work, or something to improve? Adding `!important` in the browser devtools fixes it, but I don't know how I should change in tailwind.config.js to get it to work.

**Comment by simonswiss** [comment, marked as answer]
Hey!

This nested `ul` vertical spacing for the `last-child` is by design! You can see the code responsible for it here:

https://github.com/tailwindlabs/tailwindcss-typography/blob/master/src/styles.js#L302-L314

<img src="https://user-images.githubusercontent.com/485747/127579750-60bc5ec6-0672-465f-8820-70826a17cc21.png" width="480" alt="" />

If you want to override this behaviour, you can extend the `typography` object in your Tailwind config:

```diff
theme: {
  extend: {
+    typography: { // your customisations here }
  }
}
```

Inside there, you can target default styles with the `DEFAULT` key, or specific breakpoints, as specified in the [Typography plugin documentation](https://github.com/tailwindlabs/tailwindcss-typography#customization).

Here's a little example I made – you can see I target the `ul > li > :last-child to override it, but you can style any selector this way – you may need to spend some time figuring out exactly what behaviour you want here.

https://play.tailwindcss.com/NWJZJOaZiP?file=config

![image](https://user-images.githubusercontent.com/485747/127579922-f6c63e7b-4b56-4f13-8b51-3618b5506a0b.png)

Here's the same example without the customisation, so you can see the margin-bottom applied to the nested `ul`:

![image](https://user-images.githubusercontent.com/485747/127579944-a404d39e-43f3-45ff-bdf4-e93442f92e54.png)

Hope it helps!

**Reply by aschrijver** [reply]
Thanks a heap for providing this answer!

PS. I thoroughly enjoy the whole Tailwind experience, and the typography plugin is great for making light work in (markdown-based) static site generators. Thank you for all you efforts :pray:

**Reply by simonswiss** [reply]
Glad I could help, and thank you for your kind words ✨
